pecos/core/utils/mmap_hashmap.hpp

```diff
@@ -228,6 +229,413 @@ class AnkerlStr2IntMmapableVector {
+            friend class iter_t;
+
+            public:
+                using iterator_category = std::forward_iterator_tag;
+                using difference_type = AnkerlFixedLenStr2IntMmapableVector::difference_type;
+                using value_type = AnkerlFixedLenStr2IntMmapableVector::value_type;
+                using reference = typename std::conditional_t<IsConst,
+                    value_type const&, value_type&>;
+                using pointer = typename std::conditional_t<IsConst,
+                    AnkerlFixedLenStr2IntMmapableVector::const_pointer, AnkerlFixedLenStr2IntMmapableVector::pointer>;
+
+                iter_t() noexcept = default;
+
+                template <bool OtherIsConst, typename = typename std::enable_if<IsConst && !OtherIsConst>::type>
+                constexpr iter_t(iter_t<OtherIsConst> const& other) noexcept
+                    : iter_data_(other.iter_data_) {}
+
+                constexpr iter_t(ptr_t data) noexcept
+                    : iter_data_(data) {}
+
+                template <bool OtherIsConst, typename = typename std::enable_if<IsConst && !OtherIsConst>::type>
+                constexpr auto operator=(iter_t<OtherIsConst> const& other) noexcept -> iter_t& {
+                    iter_data_ = other.iter_data_;
+                    return *this;
+                }
+
+                constexpr auto operator++() noexcept -> iter_t& {
+                    ++iter_data_;
+                    return *this;
+                }
+
+                constexpr auto operator+(difference_type diff) noexcept -> iter_t {
+                    return {iter_data_ + diff};
+                }
+
+                template <bool OtherIsConst>
+                constexpr auto operator-(iter_t<OtherIsConst> const& other) noexcept -> difference_type {
+                    return static_cast<difference_type>(iter_data_ - other.iter_data_);
+                }
+
+                constexpr auto operator*() const noexcept -> reference {
+                    return *iter_data_;
+                }
+
+                constexpr auto operator->() const noexcept -> pointer {
+                    return iter_data_;
+                }
+
+                template <bool O>
+                constexpr auto operator==(iter_t<O> const& o) const noexcept -> bool {
+                    return iter_data_ == o.iter_data_;
+                }
+
+                template <bool O>
+                constexpr auto operator!=(iter_t<O> const& o) const noexcept -> bool {
+                    return !(*this == o);
+                }
+        };
+};
+
+
+// Memory-mappable vector of std::pair<StrView, uint64_t> for Ankerl
+// This vector takes/gets std::string_view as the key, but emplace back as the special mmap format StrView
+// The key must be of length 10
+class AnkerlFixedLen10Str2IntMmapableVector {
+    template <bool IsConst>
+    class iter_t;
+
+    static constexpr std::size_t fixed_str_len = 10;
+
+    struct StrView {
```

Comment:
Should rename this struct into something like `FixLen10Str`. It's not a "view" anymore, but actually holds data.